modules/nlu-testing/src/views/full/TestModal.tsx

```diff
@@ -41,64 +30,34 @@ export const TestModal: FC<Props> = props => {
         .uniq()
         .value()
       setAvailableCtxs([...ctxs, TEST_ALL_CTX, NONE_CTX])
-      setExpectedCtx(ctxs[0])
     })
   }, [])
 
   useEffect(() => {
     const { test } = props
-    
-    setIsEditing(!!test)
 
-    if (!test && utterance !== '') {
-      setUtterance('')
-      setTestingCtx(TEST_ALL_CTX)
-      setExpectedCtx(NONE_CTX)
-      setExpectedIntent(NONE_INTENT)
-      setSlotConditions({})
+    if (!test && state.utterance !== '') {
+      dispatch({ type: 'resetState' })
     }
 
     if (test) {
-      setStateFromTest(test)
+      dispatch({ type: 'setStateFromTest', data: { test: props.test, intents } })
     }
   }, [props.test])
 
-  const setStateFromTest = (test: Test) => {
-    setUtterance(test.utterance)
-    setTestingCtx(test.context)
-
-    if (test.context === TEST_ALL_CTX) {
-      const ctxCondition = test.conditions.find(([key]) => key === 'context')
-      
-      setExpectedCtx(ctxCondition[2])
-    }
-
-    const intentCondition = test.conditions.find(([key]) => key === 'intent')
-    const expectedIntent = intents.find(i => i.name === intentCondition[2])
-    
-    setExpectedIntent(expectedIntent)
-
-    const expectedSlots = test.conditions
-      .filter(([key]) => key.startsWith('slot'))
-      .reduce((slotsDic, [key, is, slotValue]) => {
-        const slotName = key.split(':')[1]
-        slotsDic[slotName] = slotValue
-        return slotsDic
-      }, {})
-    setSlotConditions(expectedSlots)
-  }
+  const isEditing = () => !!props.test
 
   const createTest = async e => {
     e.preventDefault()
 
     const test: Test = {
-      id: isEditing ? props.test.id : Date.now().toString(),
-      utterance: utterance,
-      context: testingCtx,
+      id: isEditing() ? props.test.id : Date.now().toString(),
+      utterance: state.utterance,
```

Comment:
You could deconstruct the state object before the rest of the logic so you don't have to add `state.` everywhere
```
const { utterance, testingCtx, etc... } = state
```